h2o-r/h2o-package/R/import.R

```diff
@@ -78,23 +79,25 @@
 #' @export
 h2o.importFile <- function(path, destination_frame = "", parse = TRUE, header=NA, sep = "", col.names=NULL,
                            col.types=NULL, na.strings=NULL, decrypt_tool=NULL, skipped_columns=NULL,
-                           custom_non_data_line_markers=NULL, partition_by=NULL) {
+                           custom_non_data_line_markers=NULL, partition_by=NULL, quotechar=NULL) {
   h2o.importFolder(path, pattern = "", destination_frame=destination_frame, parse, header, sep, col.names, col.types,
                    na.strings=na.strings, decrypt_tool=decrypt_tool, skipped_columns=skipped_columns,
-                   custom_non_data_line_markers=custom_non_data_line_markers, partition_by)
+                   custom_non_data_line_markers=custom_non_data_line_markers, partition_by, quotechar)
 }
 
 
 #' @rdname h2o.importFile
 #' @export
 h2o.importFolder <- function(path, pattern = "", destination_frame = "", parse = TRUE, header = NA, sep = "",
                              col.names = NULL, col.types=NULL, na.strings=NULL, decrypt_tool=NULL, skipped_columns=NULL,
-                             custom_non_data_line_markers=NULL, partition_by=NULL) {
+                             custom_non_data_line_markers=NULL, partition_by=NULL, quotechar=NULL) {
```

Comment:
in R, I think we can directly enforce supported values in the signature:
```
quotechar=c("'", '"')
```

Reply:
1. Should be `c(NULL, "'", '"')`, provided this was possible.
2. This sets the default to the vector of these elements, nothing else. No validation effect observed.

Reply:
yes sorry, validation is not done automatically, you need to apply `match.arg`.
Anyway, this is idiomatic R apparently (we use it in various places in the code
```
> fun <- function(p=c("foo", "bar")) {
+     p <- match.arg(p)
+     p
+ }
> fun("foo")
[1] "foo"
> fun(NULL)
[1] "foo"
> fun("baz")
Error in match.arg(p) : 'arg' should be one of “foo”, “bar”
```

Reply:
so in this case, you can just put the default valud first in the propositions

Reply:
Sorry, doesn't work with `NULL` in the vecr for me:

```
h2o.importFile <- function(path, destination_frame = "", parse = TRUE, header=NA, sep = "", col.names=NULL,
                           col.types=NULL, na.strings=NULL, decrypt_tool=NULL, skipped_columns=NULL,
                           custom_non_data_line_markers=NULL, partition_by=NULL, quotechar=c(NULL, "'", '"')) {
  quotechar <- match.arg(quotechar)
```

It will automatically use the first non-null item.

```
> h2o.importFile(path = "/home/pavelp/Downloads/single_quotes_mixed.csv")
Called from: eval(expr, p)
Browse[1]> n
debug at ~/projects/h2o-3/h2o-r/h2o-package/R/import.R#84: h2o.importFolder(path, pattern = "", destination_frame = destination_frame, 
    parse, header, sep, col.names, col.types, na.strings = na.strings, 
    decrypt_tool = decrypt_tool, skipped_columns = skipped_columns, 
    custom_non_data_line_markers = custom_non_data_line_markers, 
    partition_by, quotechar)
Browse[2]> quotechar
[1] "'"
```

![Screenshot from 2020-11-27 16-28-29](https://user-images.githubusercontent.com/8769110/100464288-a0416000-30cd-11eb-90df-d3b869479863.png)

If that's a big problem, can we stick to the form of validation that is already present there and the rest of the method uses it ? I'm not against improvements, but this doesn't seem to be very functional and the rest of the surrounding code uses the same validation anyway.

Reply:
you don't need `NULL` in the `c()`, you can just put the double quote (default) first in the suggestions

Reply:
I do need NULL because that's the maybe-in-future-to-be-done AUTO mode ? Sure, I don't need it RIGHT NOW, but then it would have to be re-written in future anyway. Which it probably won't, so you're technically correct.

Reply:
well, if one day we have an "auto" mode, we can always call it "auto" instead of NULL and put it first, as NULL will automatically be converted to "auto"...

Reply:
Which removes the possibility to use "auto" as a quote :D Sure, we might do that. I was just sticking to the defaulte behavior. I don't care. We might adjust the Python client as well.

Reply:
Still, the parameter is named `quotechar`. Logically, when quotechar is set to anything, it means the quotechar is that thing set. If it's null, it means no quotechar is set explicitly and the system should decide. But my logic might be too exact.

Reply:
I don't want to discuss the sex of angels here, as I think that both approaches (NULL/None vs. 'auto') are valid given that no one would seriously consider 'auto', first as a char, second as a delimiter, without mentioning that the method documentation explicitely mentions which characters are allowed.

My point is just about using idiomatic R when possible and h2o code tends to use `c(...)`-style of param whenever the value choices are predefined.

Reply:
Leaving that to None in Python and using AUTO in R makes most sense to me then.